src/ruby/action/system.rb

```diff
@@ -62,8 +62,17 @@ def run
       }        
       begin
         if(@su)
-          puts   "#{@su} '#{@programName} #{processed_arg.join(' ')}' 2>&1" if $DEBUG 
-          @response =  `#{@su} '#{@programName} #{processed_arg.join(' ')}' 2>&1` 
+          puts   "#{@su} '#{@programName} #{processed_arg.join(' ')}'" if $DEBUG
+          # There is an issue on Ubuntu16 where a service is stopped it is not 
+          # releasing stdout or stderr file handle correctly. So as a workaround to that
+          # output is redirected to a temp file and then temp file read for a response
+          @response =  `#{@su} '#{@programName} #{processed_arg.join(' ')}' >/tmp/output 2>>/tmp/output`
```

Comment:
Hard coding /tmp/output isn't secure, it would be nice to do better (mktemp).
Ref: http://www.linuxsecurity.com/content/view/115462/151/

Also, for redirecting stderr to stdout it would be preferred to tie the filehandle together via 2>&1

Lastly, the output file is created regardless of success or failure so should probably remove it regardless of file size.

Reply:
You are right. The preferred way would be to tie file handles. But on Ubuntu16 machines I am facing this weird issue - when ZCS service is started after it was stopped it fails to release file handle for STDOUT/STDERR correctly and that is why I have made this as a workaround.